app/models/concerns/taxonomix.rb

```diff
@@ -87,10 +87,17 @@ def taxable_ids(loc = which_location, org = which_organization, inner_method = w
       inner_ids
     end
 
+    # taxonomy can be either specific taxonomy object or array of these objects
+    # it can also be an empty array that means all taxonomies (user is not assigned to any)
     def inner_select(taxonomy, inner_method = which_ancestry_method)
       # always include ancestor_ids in inner select
-      taxonomy_ids = (taxonomy.send(inner_method) + taxonomy.ancestor_ids).uniq
-      TaxableTaxonomy.where(:taxable_type => self.name, :taxonomy_id => taxonomy_ids).pluck(:taxable_id).compact.uniq
+      conditions = { :taxable_type => self.name }
+      if taxonomy.present?
+        taxonomy_ids = Array(taxonomy).map { |t| t.send(inner_method) + t.ancestor_ids }.flatten.uniq
```

Comment:
I had same feeling, just didn't know where to put it. It's in Taxonomix, Organization and Location. Maybe it could be Taxonomy class method? (I just don't want to introduce new concern).